.github/workflows/cibuild-setup-ubuntu.sh

```diff
@@ -41,7 +41,9 @@ PACKAGES_OPTIONAL=(
 if [[ "$QEMU_USER" != "1" ]]; then
 	MODULES_PACKAGE="linux-modules-extra-$(uname -r)"
 	# may not exist anymore
-	if apt-cache show "$MODULES_PACKAGE" >/dev/null 2>&1; then
+	APT_CACHE_OUTPUT=$(apt-cache show "$MODULES_PACKAGE")
+	APT_CACHE_STATUS=$?
+	if [[ "${APT_CACHE_STATUS}" == 0 && -n "${APT_CACHE_OUTPUT}" ]]; then
 		PACKAGES+=("$MODULES_PACKAGE")
 	fi
```

Comment:
There is still something wrong with 
```
Reading package lists...
+ PACKAGES=(bc btrfs-progs dnsutils libcap-ng-dev libncursesw5-dev libpam-dev libudev-dev gtk-doc-tools mdadm ntp socat asciidoctor meson lcov gpg-agent git squashfs-tools iproute2 dmsetup python3-dev gawk)
+ PACKAGES_OPTIONAL=(libsystemd-daemon-dev libsystemd-journal-dev)
+ [[ '' != \1 ]]
++ uname -r
+ MODULES_PACKAGE=linux-modules-extra-5.15.0-1042-azure
++ apt-cache show linux-modules-extra-5.15.0-1042-azure
E: No packages found
+ APT_CACHE_OUTPUT=
Error: Process completed with exit code 100.
```

Reply:
I missed set -e is done at the beginning of the script.
I'm trying the revised version.

Reply:
(Yes, "-e"/pipefail  ... that's what I'll dream about tomorrow night ...)

I'll wait for your branch to pass CI to get a proper bugfix for the master branch.

Reply:
See #2438 about this topic.